azurerm/resource_arm_storage_share_directory.go

```diff
@@ -0,0 +1,189 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/file/directories"
+)
+
+func resourceArmStorageShareDirectory() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmStorageShareDirectoryCreate,
+		Read:   resourceArmStorageShareDirectoryRead,
+		Update: resourceArmStorageShareDirectoryUpdate,
+		Delete: resourceArmStorageShareDirectoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Should we validate this with at least `NoEmptyStrings`?

Reply:
👍

> File share names can contain only lowercase letters, numbers, and hyphens, and must begin and end with a letter or a number. The name cannot contain two consecutive hyphens.